### PR TITLE
[14.0] [mig] hr_timesheet_task_domain

### PR DESCRIPTION
**Pull request by fshah-initos**

### CODE REVIEW

**Review by dsolanki-initos — APPROVED** [review]
LGTM. Functional and code review.

**Comment by fshah-initos** [comment]
@Tardo It would be great if you could add reviews here

File: hr_timesheet_task_domain/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Copyright 2020 Tecnativa - Manuel Calero
+# Copyright 2020 CorporateHub (https://corporatehub.eu)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Task Log: limit Task by Project",
+    "version": "14.0.1.0.0",
+    "category": "Human Resources",
+    "website": "https://github.com/OCA/timesheet",
+    "author": "Tecnativa, " "CorporateHub, " "Odoo Community Association (OCA)",
```
Comment (Tardo) [inline hr_timesheet_task_domain/__manifest__.py]:
```suggestion
    "author": "Tecnativa, CorporateHub, Odoo Community Association (OCA)",
```

File: hr_timesheet_task_domain/models/account_analytic_line.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.onchange("project_id")
+    def _onchange_project_id(self):
+        # Check if 'closed' field exists (provided by project_stage_closed)
+        project_stage_closed = (
+            "project.task.type" in self.env
+            and "is_closed" in self.env["project.task.type"]._fields
+        )
```
Comment (Tardo) [inline hr_timesheet_task_domain/models/account_analytic_line.py]: See https://github.com/OCA/project/pull/813
```suggestion
```

File: hr_timesheet_task_domain/models/account_analytic_line.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        task = self.task_id
+        res = super()._onchange_project_id()
+        if res is None:
+            res = {}
+        if self.project_id:  # Show only opened tasks
+            task_domain = [("project_id", "=", self.project_id.id)]
+            if project_stage_closed:
+                task_domain = task_domain + [("stage_id.is_closed", "=", False)]
```
Comment (Tardo) [inline hr_timesheet_task_domain/models/account_analytic_line.py]:
```suggestion
            task_domain = [("project_id", "=", self.project_id.id), ("stage_id.is_closed", "=", False)]
```

**Review by Tardo — CHANGES_REQUESTED** [review]
Thanks, please check the comments.

**Comment by fshah-initos** [comment]
It would be nice if we get the reviews here

**Comment by fshah-initos** [comment]
It would be nice if we get the reviews here

**Comment by fshah-initos** [comment]
It would be nice if we get the reviews here

**Review by ivs-cetmix — APPROVED** [review]
Looks good

**Comment by fshah-initos** [comment]
Hi @Tardo can you please check the latest changes

**Comment by fshah-initos** [comment]
It would be nice if we could merge this PR

**Comment by fshah-initos** [comment]
Is it possible to merge this PR?

**Review by FernandoRomera — APPROVED** [review]

**Review by AntoniRomera — APPROVED** [review]

**Comment by AntoniRomera** [comment]
@Tardo Can you check the PR?

**Comment by AntoniRomera** [comment]
Hello @pedrobaeza  

You can make the merge of this PR to be able to continue with the migration of the "project_timsheet_time_control" to be able to make the merge of the modules "helpdesk_timesheet" and "helpdesk_timesheet_time_control" as you have proposed, I have made the PR to be able to have all the options and to see which is preferred, but to make the merge more dependencies are needed.

**Pull request merged**